src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts

```diff
@@ -156,41 +160,57 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
     }
   }
 
-  isChecked(): boolean {
-    return this.checkedCities.length === this.cities.length;
+  public onSelectStation(event: MatAutocompleteSelectedEvent): void {
+    const newValue = event.option.value;
+    if (this.selectedStation.includes(newValue)) {
+      this.selectedStation = [...this.selectedStation.filter((item) => item !== newValue)];
```

Comment:
it seems spread operator here not needed)

this.selectedStation = this.selectedStation.filter((item) => item !== newValue)  - it should work